scripts/logs-backup-from-all-pods/extract_logs.py

```diff
@@ -0,0 +1,187 @@
+import datetime
+import logging
+import subprocess
+import sys
+import json
+
+import psycopg2
+import redis
+import requests
+from datetime import datetime, timedelta
+
+def login_to_isd():
+    try:
+        cookie = ""
+        cmd = "curl -vvv -X POST '" + isd_gate_url + "/login?username=" + isd_admin_username + "&password=" + isd_admin_password + "&submit=Login'"
+        output = subprocess.getoutput(cmd=cmd)
+        output = output.replace(isd_admin_username, "***").replace(isd_admin_password, "***")
+        logging.info(f"Output for ISD login : {output}")
+        components = output.split("<")
+        for comp in components:
+            if comp.__contains__("set-cookie") or comp.__contains__("Set-Cookie"):
+                cookie = comp.split(":")[1].strip()
+        return cookie
+    except Exception as e:
+        print("Exception occurred while logging in to ISD : ", e)
+        logging.error("Exception occurred while logging in to ISD : ", exc_info=True)
+        raise e
+
+
+def get_response(url, cookie):
+    try:
+        logging.info(url)
+        headers = {'cookie': cookie, 'x-spinnaker-user': isd_admin_username}
+        request = requests.get(url=url, headers=headers)
+        logging.info(f"getResponse response : {request}")
+        return request.json()
+    except Exception as e:
+        print("Exception occurred while fetching the gate response: ", e)
+        logging.error("Exception occurred while fetching the gate response: ", exc_info=True)
+        raise e
+
+
+def get_pipelines_executed_users_count():
+    keys = redis_conn.keys("pipeline:*:stageIndex")
+    names = set()
+    if days.strip() != 'None':
+        pl_key_exec_dict = {}
+        for key in keys:
+            executions = redis_conn.lrange(key, 0, -1)
+            plKey = key[:35]
+            pl_key_exec_dict[plKey] = executions[len(executions) - 1]
+        # print("The pl_key_exec_dict is: ", pl_key_exec_dict)
+        date = datetime.datetime.now() - datetime.datetime(1970, 1, 1)
```

Comment:
Whats this ? why are we subtracting the date from 1970 ?

Reply:
that's the standard way to get the epoch format of the start date